autoload/jedi.vim

```diff
@@ -385,19 +413,47 @@ endfunction
 " ------------------------------------------------------------------------
 
 function! jedi#add_goto_window(len) abort
-    set lazyredraw
-    cclose
     let height = min([a:len, g:jedi#quickfix_window_height])
-    execute 'belowright copen '.height
-    set nolazyredraw
+    " " Q: why the cclose?  To get the height always?
+    " " Should save/restore lazyredraw otherwise.
+    " let save_lazyredraw = &lazyredraw
+    " set lazyredraw
+    " try
+    "     execute 'belowright cwindow '.height
+    " finally
+    "     let &lazyredraw = save_lazyredraw
+    " endtry
+
+    " Using :cwindow allows to stay in the current window in case it is opened
+    " already.
+    execute 'belowright cwindow '.height
+
     if g:jedi#use_tabs_not_buffers == 1
         noremap <buffer> <CR> :call jedi#goto_window_on_enter()<CR>
     endif
+
+    function! s:on_closed_qf() abort
+      " Remove global usages when the quickfix buffer is removed from the
+      " window, e.g. with :cclose.
+      PythonJedi jedi_vim.highlight_usages(None)
```

Comment:
I think if you do this, I get a lot of `jedi-vim: called highlight_usages_for_win without highlights!`

Note that I use `g:jedi#use_tabs_not_buffers == 1`.

I'm not sure why this happens. However since it generally doesn't really clean up usages for me (neither with TextChanged, nor InsertEnter), I think there's something fundamentally wrong with the cleanup, but I don't really understand what it is.

Reply:
This should be improved/fix, and/or needs a revisit again.